sky/utils/kubernetes/kind_port_config.py

```diff
@@ -0,0 +1,112 @@
+            }
+        }
+        config_path = os.path.expanduser(CONFIG_PATH)
+        common_utils.dump_yaml(config_path, default_config)
+        skypilot_config.try_load_config()
+
+    logger.debug(f'Creating backup at {BACKUP_PATH} of'
+                 'current kubernetes port forwarding option.')
+    backup_path = os.path.expanduser(BACKUP_PATH)
+    backup_port_config = skypilot_config.get_nested(
+        ('kubernetes', 'ports'),
+        kubernetes_enums.KubernetesPortMode.LOADBALANCER.value)
+    common_utils.dump_yaml(backup_path,
+                           {'kubernetes': {
+                               'ports': backup_port_config
+                           }})
+
+    current_path = skypilot_config.loaded_config_path()
+    logger.debug(
+        f'Set current port forwarding option to ingress in {current_path}.')
+    updated_config = skypilot_config.set_nested(
+        ('kubernetes', 'ports'),
+        kubernetes_enums.KubernetesPortMode.INGRESS.value)
+    common_utils.dump_yaml(current_path, updated_config)
+    skypilot_config.try_load_config()
+
+
+def restore_port_config():
+    """Restore the original value of `kubernetes:ports`.
+
+    Restores the value of `kubernetes:ports` from ~/.sky/backup_config.yaml
+    in the active config file. Then, clears the ~/.sky/backup_config.yaml
+    file.
+    """
+
+    backup_path = os.path.expanduser(BACKUP_PATH)
+    try:
+        backup_port_config = common_utils.read_yaml(
+            backup_path)['kubernetes']['ports']
+    except FileNotFoundError:
+        logger.debug(f'Could not find backup file at {BACKUP_PATH}.')
+        return
+    except (TypeError, KeyError):
+        logger.debug(f'Could not find `kubernetes:ports` in {BACKUP_PATH}.')
+        return
+
+    current_path = skypilot_config.loaded_config_path()
+    if current_path is None:
+        logger.debug('There is no current active config file.')
+        return
+
+    logger.debug(f'Restore original port forwarding option in {current_path}'
+                 f'from {BACKUP_PATH}.')
+    restored_config = skypilot_config.set_nested(('kubernetes', 'ports'),
+                                                 backup_port_config)
+    common_utils.dump_yaml(current_path, restored_config)
+    logger.debug(f'Resetting {BACKUP_PATH} to be empty.')
+    common_utils.dump_yaml(backup_path, {})
+    skypilot_config.try_load_config()
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--modify',
+        action='store_true',
+        help='Backup current port config and set current value to ingress')
+    parser.add_argument('--restore',
+                        action='store_true',
+                        help='Restore the port config from backup')
+    args = parser.parse_args()
+
+    if args.modify:
+        modify_port_config()
+    elif args.restore:
+        restore_port_config()
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
Tried running `sky local up` with just some commented text in my `~/.sky/config.yaml, but it failed.

My `~/.sky/config.yaml`:
```
#kubernetes:
#  pod_config:
#    spec:
#      imagePullSecrets:
#        - name: regcred
```

`sky local up` failed:
```
# My ~/sky_logs/sky-2024-03-23-23-00-22-831320/local_up.log
No kind clusters found.
Generating /tmp/skypilot-kind.yaml
D 03-23 23:00:23 skypilot_config.py:136] Using config path: /Users/romilb/.sky/config.yaml
D 03-23 23:00:23 skypilot_config.py:140] Config loaded:
D 03-23 23:00:23 skypilot_config.py:140] None
D 03-23 23:00:23 skypilot_config.py:150] Config syntax check passed.
Creating cluster "skypilot" ...
 • Ensuring node image (kindest/node:v1.25.3) 🖼  ...
 ✓ Ensuring node image (kindest/node:v1.25.3) 🖼
 • Preparing nodes 📦   ...
 ✓ Preparing nodes 📦 
 • Writing configuration 📜  ...
 ✓ Writing configuration 📜
 • Starting control-plane 🕹️  ...
 ✓ Starting control-plane 🕹️
 • Installing CNI 🔌  ...
 ✓ Installing CNI 🔌
 • Installing StorageClass 💾  ...
 ✓ Installing StorageClass 💾
Set kubectl context to "kind-skypilot"
You can now use your cluster with:

kubectl cluster-info --context kind-skypilot

Have a question, bug, or feature request? Let us know! https://kind.sigs.k8s.io/#community 🙂
Kind cluster created.
Pulling SkyPilot CPU image...
latest: Pulling from skypilot-375900/skypilotk8s/skypilot
Digest: sha256:eb8f58d9ced1a7b64269fca7e04ad0576f4b0a3f3b6f271f23399b12347dde76
Status: Image is up to date for us-central1-docker.pkg.dev/skypilot-375900/skypilotk8s/skypilot:latest
us-central1-docker.pkg.dev/skypilot-375900/skypilotk8s/skypilot:latest
Loading SkyPilot CPU image into kind cluster...
Image: "" with ID "sha256:d3d7e123a4a1743423a05683fb21f5348c5718773416f519575391310d345f10" not yet present on node "skypilot-control-plane", loading...
SkyPilot CPU image loaded into kind cluster.
Updating Kubernetes port configuration...
D 03-23 23:01:08 skypilot_config.py:136] Using config path: /Users/romilb/.sky/config.yaml
D 03-23 23:01:08 skypilot_config.py:140] Config loaded:
D 03-23 23:01:08 skypilot_config.py:140] None
D 03-23 23:01:08 skypilot_config.py:150] Config syntax check passed.
Traceback (most recent call last):
  File "/Users/romilb/tools/anaconda3/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/romilb/tools/anaconda3/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/utils/kubernetes/kind_port_config.py", line 112, in <module>
    main()
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/utils/kubernetes/kind_port_config.py", line 106, in main
    modify_port_config()
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/utils/kubernetes/kind_port_config.py", line 53, in modify_port_config
    updated_config = skypilot_config.set_nested(
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/skypilot_config.py", line 102, in set_nested
    _check_loaded_or_die()
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/skypilot_config.py", line 165, in _check_loaded_or_die
    raise RuntimeError(
RuntimeError: No user configs loaded. Check ~/.sky/config.yaml exists and can be loaded.
```